torchrl/data/replay_buffers/samplers.py

```diff
@@ -57,6 +59,10 @@ def ran_out(self) -> bool:
         # by default, samplers never run out
         return False
 
+    @abstractmethod
+    def _empty(self):
+        ...
```

Comment:
`raise NotImplementedError` would be more consistent with `Sampler.sample`.

Reply:
I noticed below you're actually favouring Ellipsis over NotImplementedError so perhaps we should change `sample` instead?

Reply:
Sure, I'm happy with that.